### ISSUE TITLE
Crash in MPWebSocket.m:1277 (call to assertOnWorkQueue)

### ISSUE THREAD

**Issue by dgatwood** [mixpanel/mixpanel-iphone#261]
I just saw a fascinating crash report in MixPanel code that occurred only about 1.5 seconds after our app launched.  Apparently, assertOnWorkQueue failed even though the call path very clearly showed that the code was running within that work queue....

The only thing I can imagine is that the MPWebSocket object was getting deallocated while the code was running (ick) and therefore the call to dispatch_get_specific was comparing a newly nil self pointer with an arbitrary void pointer value from the dispatch queue that wouldn't have gotten zeroed.  If I'm right, then this is probably just another manifestation of https://github.com/mixpanel/mixpanel-iphone/issues/234 caused by the MPWebSocket object going away while other code is still actively using it.  However, I can't be certain, hence I'm filing a separate ticket.

The backtrace is shown below.

<pre>
Thread : Crashed: Thread
0  libsystem_kernel.dylib         0x0000000197a4f270 __pthread_kill + 8
1  libsystem_pthread.dylib        0x0000000197aed170 pthread_kill + 112
2  libsystem_c.dylib              0x00000001979c6b18 abort + 112
3  libsystem_c.dylib              0x000000019799ba3c basename
4  HotspotVPN                     0x000000010012a434 -[MPWebSocket assertOnWorkQueue] (MPWebSocket.m:368)
5  HotspotVPN                     0x0000000100132154 -[MPWebSocket _pumpScanner] (MPWebSocket.m:1277)
6  HotspotVPN                     0x00000001001332a8 __34-[MPWebSocket stream:handleEvent:]_block_invoke523 (MPWebSocket.m:1432)
7  libdispatch.dylib              0x0000000197909994 _dispatch_call_block_and_release + 24
8  libdispatch.dylib              0x0000000197909954 _dispatch_client_callout + 16
9  libdispatch.dylib              0x00000001979140a4 _dispatch_queue_drain + 1448
10 libdispatch.dylib              0x000000019790ca5c _dispatch_queue_invoke + 132
11 libdispatch.dylib              0x0000000197916318 _dispatch_root_queue_drain + 720
12 libdispatch.dylib              0x0000000197917c4c _dispatch_worker_thread3 + 108
13 libsystem_pthread.dylib        0x0000000197ae922c _pthread_wqthread + 816
</pre>

**Comment by samgreen** [comment]
Waiting on #262 review

**Comment by samgreen** [comment]
This will be resolved in the next release.
